### ISSUE TITLE
performance issue

### ISSUE THREAD

**Issue by liuhongbo** [adampietrasiak/jquery.initialize#1]
I think there is a performance issue with this implementation.
the code 

>`$.each(mutation.addedNodes, function(i, addedNode){
							target = target.add(addedNode);
						});`

will have very bad performance when the addedNodes length goes to thousands.
the performance might related to jquery, but we can void by just use a pure javascript array instead of the jquery object to hold the target.

You can repeat this issue at 
[https://jsfiddle.net/liuhongbo/f8moz1cz/](https://jsfiddle.net/liuhongbo/f8moz1cz/)

**Comment by jaredplanter** [comment]
I am running into this exact problem in my application right now. 

Are there plans to make the proposed change? If not, I can put up a pull request.

**Comment by adampietrasiak** [comment]
I don't have a lot of time right now, so if you have a while - PR would be welcomed!

**Comment by dbezborodovrp** [comment]
I tried replacing `$.each` with traditional for loops, which did not improve performance (as was expected; benchmarks out there indicate that jQuery is not much slower than traditional for loops anyhow.)

I believe it is a deeper underlying issue. I don't think that addedNodes.length should ever get into the thousands if you are doing simple insertions of just one DOM node at a time. For now, I wrote my own observer class, which solved my performance issues (the page crashed after using 4 initialize() selector, where as now it is lightning fast.
